playground/src/pages/CodingEnvironment.tsx

```diff
@@ -214,9 +212,33 @@ Cycles = [${trace_length.toString()}]`);
       toast.error("Not yet available");
     });
   };
+
+  /**
+   * This verifies the proof that is stored in the session. 
+   * It runs the Rust program that is imported above.
+   * It returns true if the proof is valid, and false otherwise.
+   * As inputs we need program_info, stack_input, stack_output, and proof.
+   */
   const verifyProgram = async () => {
     init().then(() => {
-      toast.error("Not yet available");
+      try {
+
+        if (checkInputs(inputs)) {
+          const result = verify_program(code, inputs, output, proof);
```

Comment:
it looks like you're assuming that both the output and the proof are available. We should check for these just like we check inputs, and then raise a toast message if they're missing telling people to prove a program first

Reply:
did it